cmake/Modules/FindComputeCpp.cmake

```diff
@@ -308,7 +310,7 @@ function(__build_ir)
 
   if(NOT MSVC)
     # Add a custom target for the generated integration header
-    add_custom_target(${headerTargetName} DEPENDS ${outputSyclFile})
+    add_custom_target(${headerTargetName} DEPENDS ${outputDeviceFile} ${outputSyclFile})
```

Comment:
The object file isn't really a dependency of the header target.

Reply:
Kind of, the content of the object file is in the sycl file :P 

More rational, this allow automatic regeneration of the object file, and if one file needs regeneration then the other needs it as well.

Reply:
It is not consumed by the SYCL program. Therefore it does not depend on it.

There's nothing really *wrong* here, it just doesn't feel clean.

Side note: you could make the object file a `BYPRODUCT` instead of an `OUTPUT` if it only serves debug purposes.

> More rational, this allow automatic regeneration of the object file, and if one file needs regeneration then the other needs it as well.

With this change the SYCL translation unit is rebuilt when the object file or the integration header changes. Since integration header and object file are built by the same command, there will be no semantic difference.